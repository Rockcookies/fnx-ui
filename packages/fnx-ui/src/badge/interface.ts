import { ReactNode, HTMLAttributes } from 'react';

export type BadgeProps = BadgeComponentProps &
	Omit<HTMLAttributes<HTMLSpanElement>, 'color'>;

export interface BadgeComponentProps {
	count?: ReactNode;
	showZero?: boolean;
	dot?: boolean;
	max?: string | number;
	color?: string;
	offset?: [number | string, number | string];
}
