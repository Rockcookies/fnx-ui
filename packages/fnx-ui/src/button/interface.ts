import { ButtonHTMLAttributes, ReactNode } from 'react';

export type ButtonType =
	| 'default'
	| 'primary'
	| 'success'
	| 'warning'
	| 'danger';

export type ButtonSize = 'lg' | 'md' | 'sm' | 'xs';

export type ButtonShape = 'rect' | 'radius' | 'round';

export interface ButtonComponentProps {
	type?: ButtonType;
	size?: ButtonSize;
	icon?: ReactNode;
	iconPosition?: 'left' | 'right';
	loading?: boolean;
	loadingIcon?: ReactNode;
	loadingChildren?: ReactNode;
	disabled?: boolean;
	hairline?: boolean;
	plain?: boolean;
	shape?: ButtonShape;
	block?: boolean;
	color?: string;
	htmlType?: ButtonHTMLAttributes<HTMLButtonElement>['type'];
}

export type ButtonProps = ButtonComponentProps &
	Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'type'>;
