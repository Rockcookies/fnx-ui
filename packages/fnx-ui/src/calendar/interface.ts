import { HTMLAttributes, ReactNode } from 'react';
import CalendarDay from './CalendarDay';
import CalendarMulti from './CalendarMulti';
import CalendarRange from './CalendarRange';

export type CalendarValue = Date | string | number;

export type CalendarDayType =
	| 'normal'
	| 'start'
	| 'start-end'
	| 'middle'
	| 'end'
	| 'selected'
	| 'multiple-middle'
	| 'multiple-selected'
	| 'disabled';

export interface CalendarDayComponentProps {
	color?: string;
	dayHeight?: string;
	date: Date;
	content?: ReactNode;
	type: CalendarDayType;
	topInfo?: ReactNode;
	bottomInfo?: ReactNode;
}

export type CalendarDayProps = CalendarDayComponentProps &
	Omit<HTMLAttributes<HTMLDivElement>, 'content'>;

export interface CalendarSlots {
	day?: (day: CalendarDayComponentProps) => ReactNode;
	subTitle?: boolean | ((date: Date) => ReactNode);
	monthMark?: boolean | ((date: Date) => ReactNode);
	monthTitle?: boolean | ((date: Date) => ReactNode);
	footer?: ReactNode;
}

export interface CalendarBaseProps {
	title?: ReactNode;
	color?: string;
	minDate?: Date | string | number;
	maxDate?: Date | string | number;
	dayHeight?: number | string;
	readonly?: boolean;
	closeIcon?: ReactNode;
	firstDayOfWeek?: number;
	slots?: CalendarSlots;
	confirmText?: ReactNode;
	confirmDisabledText?: ReactNode;
}

export type CalendarElementProps = Omit<
	HTMLAttributes<HTMLDivElement>,
	'defaultValue' | 'onChange' | 'title'
>;

export interface CalendarContainerRef {
	element: HTMLDivElement | null;
	reset: () => void;
	getValue: () => Date[] | undefined;
}

export interface CalendarRef {
	element: HTMLDivElement | null;
	reset: () => void;
	getValue: () => Date | undefined;
}

export interface CalendarComponentProps extends CalendarBaseProps {
	defaultValue?: CalendarValue;
	onChange?: (value: Date) => void;
	onConfirm?: (value?: Date) => void;
	onCancel?: () => void;
}

export type CalendarProps = CalendarComponentProps & CalendarElementProps;

export type CalendarRangeRef = CalendarContainerRef;

export interface CalendarRangeComponentProps extends CalendarBaseProps {
	defaultValue?: CalendarValue[];
	onChange?: (value: Date[]) => void;
	onConfirm?: (value?: Date[]) => void;
	onCancel?: () => void;
	allowSameDay?: boolean;
	maxSize?: number;
	onMaxSize?: () => void;
}

export type CalendarRangeProps = CalendarRangeComponentProps &
	CalendarElementProps;

export type CalendarMultiRef = CalendarRangeRef;

export interface CalendarMultiComponentProps extends CalendarBaseProps {
	defaultValue?: CalendarValue[];
	onChange?: (value: Date[]) => void;
	onConfirm?: (value?: Date[]) => void;
	onCancel?: () => void;
	maxSize?: number;
	onMaxSize?: () => void;
}

export type CalendarMultiProps = CalendarMultiComponentProps &
	CalendarElementProps;

export interface CalendarExportExtra {
	Range: typeof CalendarRange;
	Multi: typeof CalendarMulti;
	Day: typeof CalendarDay;
}
