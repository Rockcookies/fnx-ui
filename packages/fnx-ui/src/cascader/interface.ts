import { HTMLAttributes, ReactNode } from 'react';

export type CascaderValue = string | number;

export interface CascaderOption {
	label?: ReactNode;
	value?: CascaderOption;
	loading?: boolean;
	disabled?: boolean;
	children?: CascaderOption[];
	// for custom filed names
	[key: string]: any;
}

export interface CascaderDataNames {
	label?: string | string[];
	value?: string | string[];
	loading?: string | string[];
	disabled?: string | string[];
	children?: string | string[];
}

export interface CascaderComponentProps<T = CascaderOption> {
	defaultValue?: CascaderValue[];
	data?: T[];
	dataNames?: CascaderDataNames;
	onLoadData?: (options: T[]) => void;
	onConfirm?: (value: CascaderValue[], options: T[]) => void;
	onCancel?: () => void;
	onChange?: (value: CascaderValue[], options: T[]) => void;
	title?: ReactNode;
	placeholder?: ReactNode;
	swipeable?: boolean;
	closeIcon?: ReactNode;
	activeColor?: string;
}

export type CascaderProps<T = CascaderOption> = CascaderComponentProps<T> &
	Omit<HTMLAttributes<HTMLDivElement>, 'defaultValue' | 'title' | 'onChange'>;
