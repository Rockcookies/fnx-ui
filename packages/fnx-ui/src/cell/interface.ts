import { HTMLAttributes, ReactNode } from 'react';
import CellGroup from './CellGroup';

export type CellItemsAlign = 'top' | 'middle' | 'bottom';

export interface CellComponentProps {
	title?: ReactNode;
	titleProps?: HTMLAttributes<HTMLDivElement>;
	content?: ReactNode;
	contentProps?: HTMLAttributes<HTMLDivElement>;
	description?: ReactNode;
	descriptionProps?: HTMLAttributes<HTMLDivElement>;
	leftIcon?: ReactNode;
	leftIconProps?: HTMLAttributes<HTMLSpanElement>;
	rightIcon?: ReactNode;
	rightIconProps?: HTMLAttributes<HTMLSpanElement>;
	border?: boolean;
	itemsAlign?: CellItemsAlign;
	clickable?: boolean;
	size?: 'md' | 'lg';
}

export type CellProps = CellComponentProps &
	Omit<HTMLAttributes<HTMLDivElement>, 'title'>;

export interface CellGroupComponentProps {
	title?: ReactNode;
	titleProps?: HTMLAttributes<HTMLDivElement>;
	border?: boolean;
	insert?: boolean;
	bodyProps?: HTMLAttributes<HTMLDivElement>;
}

export type CellGroupProps = CellGroupComponentProps &
	Omit<HTMLAttributes<HTMLDivElement>, 'title'>;

export interface CellExportExtra {
	Group: typeof CellGroup;
}
