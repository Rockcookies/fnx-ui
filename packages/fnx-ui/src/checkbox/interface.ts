import { HTMLAttributes, ReactNode } from 'react';
import InternalCheckboxGroup from './CheckboxGroup';

export type CheckboxValue = string | number | boolean;

export interface CheckboxBaseProps {
	disabled?: boolean;
	direction?: 'vertical' | 'horizontal';
	checkedIcon?: ReactNode;
	icon?: ReactNode;
	iconSize?: string | number;
	iconPosition?: 'left' | 'right';
	iconShape?: 'square' | 'round' | 'plain';
	iconCheckedColor?: string;
	labelDisabled?: boolean;
}

export interface CheckboxComponentProps extends CheckboxBaseProps {
	value?: CheckboxValue;
	checked?: boolean;
	defaultChecked?: boolean;
	onChange?: (checked: boolean) => void;
	skipGroup?: boolean;
}

export type CheckboxProps = CheckboxComponentProps &
	Omit<HTMLAttributes<HTMLDivElement>, 'onChange' | 'value'>;

export interface CheckboxRef {
	element: HTMLDivElement | null;
	toggle: () => void;
}

export interface CheckboxGroupComponentProps<
	T extends CheckboxValue = CheckboxValue,
> extends CheckboxBaseProps {
	value?: T[];
	defaultValue?: T[];
	onChange?: (v: T[]) => void;
	maxCheckedCount?: number;
}

export type CheckboxGroupProps<T extends CheckboxValue = CheckboxValue> =
	CheckboxGroupComponentProps<T> &
		Omit<HTMLAttributes<HTMLDivElement>, 'onChange' | 'defaultValue'>;

export interface CheckboxGroupContextData extends CheckboxBaseProps {
	value?: CheckboxValue[];
	registerValue?: (value: CheckboxValue) => void;
	cancelValue?: (value: CheckboxValue) => void;
	toggleValue?: (isCheck: boolean, value: CheckboxValue) => void;
}
export interface CheckboxGroupRef {
	element: HTMLDivElement | null;
	toggleAll: (options?: { checkAll?: boolean }) => void;
}

export interface CheckboxExportExtra {
	Group: typeof InternalCheckboxGroup;
}
