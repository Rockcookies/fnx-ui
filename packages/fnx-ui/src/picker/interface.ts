import { HTMLAttributes, ReactNode } from 'react';
import PickerCascade from './PickerCascade';
import PickerMulti from './PickerMulti';

export type PickerValue = string | number;

export interface PickerDataNames {
	label?: string | string[];
	value?: string | string[];
	children?: string | string[];
	disabled?: string | string[];
}

export interface PickerOption {
	label?: ReactNode;
	value?: PickerValue;
	disabled?: boolean;
	children?: PickerOption[];
	// for custom filed names
	[key: string]: any;
}

export type PickerOptionOrValue = PickerOption | PickerValue;

export interface PickerBaseProps {
	optionHeight?: number | string;
	visibleOptionsCount?: number;
	transitionDuration?: number;
	toolbar?: ReactNode;
	toolbarPosition?: 'top' | 'bottom';
	title?: ReactNode;
	confirmButton?: ReactNode;
	cancelButton?: ReactNode;
	loading?: boolean;
	dataNames?: PickerDataNames;
}

export type PickerElementProps = Omit<
	HTMLAttributes<HTMLDivElement>,
	'defaultValue' | 'onChange' | 'title'
>;

export interface PickerComponentProps<T = PickerOptionOrValue>
	extends PickerBaseProps {
	defaultValue?: PickerValue;
	data?: T[];
	onChange?: (value: PickerValue, option: T) => void;
	onConfirm?: (value: PickerValue, option: T) => void;
	onCancel?: () => void;
}

export type PickerProps<T = PickerOptionOrValue> = PickerComponentProps<T> &
	PickerElementProps;

export interface PickerRef<T = PickerOptionOrValue> {
	root: HTMLDivElement | null;
	getValue: () => PickerValue;
	getOption: () => T;
}

export interface PickerMultiComponentProps<T = PickerOptionOrValue>
	extends PickerBaseProps {
	defaultValue?: PickerValue[];
	data?: Array<T[]>;
	onChange?: (value: PickerValue[], options: T[]) => void;
	onConfirm?: (value: PickerValue[], options: T[]) => void;
	onCancel?: () => void;
}

export type PickerMultiProps<T = PickerOptionOrValue> =
	PickerMultiComponentProps<T> & PickerElementProps;

export interface PickerMultiRef<T = PickerOptionOrValue> {
	root: HTMLDivElement | null;
	getValue: () => PickerValue[];
	getOptions: () => T[];
}

export interface PickerCascadeComponentProps<T = PickerOption>
	extends PickerBaseProps {
	defaultValue?: PickerValue[];
	data?: T[];
	onChange?: (value: PickerValue[], options: T[]) => void;
	onConfirm?: (value: PickerValue[], options: T[]) => void;
	onCancel?: () => void;
}

export type PickerCascadeProps<T = PickerOption> =
	PickerCascadeComponentProps<T> & PickerElementProps;

export interface PickerCascadeRef<T = PickerOption> {
	root: HTMLDivElement | null;
	getValue: () => PickerValue[];
	getOptions: () => T[];
}

export interface PickerExportExtra {
	Multi: typeof PickerMulti;
	Cascade: typeof PickerCascade;
}
