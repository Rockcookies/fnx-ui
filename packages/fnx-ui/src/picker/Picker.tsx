import React, {
	forwardRef,
	ReactElement,
	useImperativeHandle,
	useMemo,
	useRef,
} from 'react';
import { ForwardRefProps } from '../utils/interface';
import usePickerProps from './hooks/use-picker-props';
import { PickerOptionOrValue, PickerProps, PickerRef } from './interface';
import PickerColumn, { PickerColumnRef } from './PickerColumn';
import PickerContainer from './PickerContainer';

const InternalPicker = forwardRef<PickerRef, PickerProps>((_props, ref) => {
	const [
		{ optionHeight, visibleOptionsCount, transitionDuration, dataNames },
		{ defaultValue, data, onChange, onConfirm, onCancel, ...restProps },
	] = usePickerProps(_props);

	const rootRef = useRef<HTMLDivElement>(null);
	const columnRef = useRef<PickerColumnRef>(null);

	const pickerRef = useMemo<PickerRef>(
		() => ({
			element: rootRef.current,
			getValue: (): any => columnRef.current?.getActiveValue(),
			getOption: (): any => columnRef.current?.getActiveOption(),
		}),
		[],
	);

	useImperativeHandle<PickerRef, PickerRef>(ref, () => pickerRef, [
		pickerRef,
	]);

	return (
		<PickerContainer
			{...restProps}
			optionHeight={optionHeight}
			visibleOptionsCount={visibleOptionsCount}
			ref={rootRef}
			onConfirm={() => {
				onConfirm &&
					onConfirm(pickerRef.getValue(), pickerRef.getOption());
			}}
			onCancel={() => {
				onCancel && onCancel();
			}}
		>
			<PickerColumn
				optionHeight={optionHeight}
				visibleOptionsCount={visibleOptionsCount}
				transitionDuration={transitionDuration}
				defaultValue={defaultValue}
				data={data}
				dataNames={dataNames}
				onChange={() => {
					onChange &&
						onChange(pickerRef.getValue(), pickerRef.getOption());
				}}
				ref={columnRef}
			/>
		</PickerContainer>
	);
});

InternalPicker.displayName = 'Picker';

const Picker = InternalPicker as <T extends PickerOptionOrValue>(
	props: ForwardRefProps<PickerProps<T>, PickerRef<any>>,
) => ReactElement;

export default Picker;
