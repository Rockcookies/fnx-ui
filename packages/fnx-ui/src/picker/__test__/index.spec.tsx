import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';
import TestsDOM from '../../../test/dom';
import TestsEvent from '../../../test/events';
import TestsReact from '../../../test/react';
import Picker from '../index';
import { PickerRef } from '../interface';

describe('<Picker/>', () => {
	const columns = ['1990', '1991', '1992', '1993', '1994', '1995'];
	const options = [
		{
			value: '值1',
			label: 'column1',
		},
		{
			value: '值2',
			label: 'column2',
		},
	];

	beforeEach(() => {
		jest.useFakeTimers();
	});

	afterEach(() => {
		jest.useRealTimers();
	});

	it('render picker base props', () => {
		const { container } = render(
			<Picker
				data={columns}
				toolbarPosition="bottom"
				title="单选标题"
				confirmButton="确定"
				cancelButton="取消"
			></Picker>,
		);
		expect(container).toMatchSnapshot();
	});

	it('render picker do not show title、conform、cancel', () => {
		const { container } = render(
			<Picker
				data={columns}
				title={false}
				confirmButton={false}
				cancelButton={false}
			></Picker>,
		);

		expect(container).toMatchSnapshot();
	});

	it('render picker defined title、cancel、confirm', () => {
		const { container } = render(
			<Picker
				data={columns}
				title={<span>defined title</span>}
				confirmButton={<span>defined confirm button</span>}
				cancelButton={<span>defined cancel button</span>}
			></Picker>,
		);

		expect(container).toMatchSnapshot();
	});

	it('set picker optionHeight、visibleOptionsCount', async () => {
		const { container } = render(
			<Picker
				data={columns}
				optionHeight={30}
				visibleOptionsCount={4}
			></Picker>,
		);

		expect(container).toMatchSnapshot();
	});

	it('simple columns confirm & cancel event', () => {
		const onConfirm = jest.fn();
		const onCancel = jest.fn();

		const { container } = render(
			<Picker
				data={columns}
				data-testid="picker"
				onConfirm={onConfirm}
				onCancel={onCancel}
			></Picker>,
		);

		const confirm = TestsDOM.mustQuerySelector(
			container,
			'.fnx-picker__confirm',
		);

		const cancel = TestsDOM.mustQuerySelector(
			container,
			'.fnx-picker__cancel',
		);

		act(() => {
			fireEvent.click(confirm);
		});

		expect(onConfirm).toHaveBeenCalledWith('1990', '1990');

		act(() => {
			fireEvent.click(cancel);
		});
		expect(onCancel).toHaveBeenCalled();
	});

	it('drag columns', async () => {
		const onValueChange = jest.fn();
		const { container } = render(
			<Picker
				data={columns}
				optionHeight={30}
				visibleOptionsCount={4}
				onChange={onValueChange}
			></Picker>,
		);

		const track = TestsDOM.mustQuerySelector(
			container,
			'.fnx-picker-column',
		);

		await TestsEvent.triggerDrag(track, [0, -200]);

		act(() => {
			jest.advanceTimersByTime(400);
		});

		expect(onValueChange).toHaveBeenCalledWith('1995', '1995');
	});

	it('get activeValue by pickerRef', async () => {
		const pickerRef = React.createRef<PickerRef>();

		render(<Picker data={options} ref={pickerRef}></Picker>);

		expect(TestsReact.mustGetRef(pickerRef).getValue()).toEqual(
			options[0].value,
		);
	});

	/* it('get activeIndex by pickerColumnRef', async () => {
		const pickerColumnRef = React.createRef<PickerColumnRef>();

		render(
			<PickerColumn
				data={options}
				ref={pickerColumnRef}
				optionHeight={30}
				visibleOptionsCount={4}
				transitionDuration={300}
			></PickerColumn>,
		);

		expect(
			pickerColumnRef.current && pickerColumnRef.current.getActiveIndex(),
		).toEqual(0);
	}); */
});
