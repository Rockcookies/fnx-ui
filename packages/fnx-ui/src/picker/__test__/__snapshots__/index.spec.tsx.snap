// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Picker/> render picker base props 1`] = `
<div>
  <div
    class="fnx-picker"
  >
    <div
      class="fnx-picker__columns"
      style="height: 264px;"
    >
      <div
        class="fnx-picker-column"
      >
        <ul
          class="fnx-picker-column__wrapper"
          style="transform: translate3d(0, 110px, 0);"
        >
          <li
            class="fnx-picker-column__option fnx-picker-column__option--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1990
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1991
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1992
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1993
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1994
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1995
            </div>
          </li>
        </ul>
      </div>
      <div
        class="fnx-picker__mask"
        style="background-size: 100% 110px;"
      />
      <div
        class="fnx-picker__frame fnx-hairline-unset--top-bottom"
        style="height: 44px;"
      />
    </div>
    <div
      class="fnx-picker__toolbar"
    >
      <button
        class="fnx-picker__cancel"
        type="button"
      >
        取消
      </button>
      <div
        class="fnx-picker__title fnx-ellipsis"
      >
        单选标题
      </div>
      <button
        class="fnx-picker__confirm"
        type="button"
      >
        确定
      </button>
    </div>
  </div>
</div>
`;

exports[`<Picker/> render picker defined title、cancel、confirm 1`] = `
<div>
  <div
    class="fnx-picker"
  >
    <div
      class="fnx-picker__toolbar"
    >
      <span>
        defined cancel button
      </span>
      <span>
        defined title
      </span>
      <span>
        defined confirm button
      </span>
    </div>
    <div
      class="fnx-picker__columns"
      style="height: 264px;"
    >
      <div
        class="fnx-picker-column"
      >
        <ul
          class="fnx-picker-column__wrapper"
          style="transform: translate3d(0, 110px, 0);"
        >
          <li
            class="fnx-picker-column__option fnx-picker-column__option--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1990
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1991
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1992
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1993
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1994
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1995
            </div>
          </li>
        </ul>
      </div>
      <div
        class="fnx-picker__mask"
        style="background-size: 100% 110px;"
      />
      <div
        class="fnx-picker__frame fnx-hairline-unset--top-bottom"
        style="height: 44px;"
      />
    </div>
  </div>
</div>
`;

exports[`<Picker/> render picker do not show title、conform、cancel 1`] = `
<div>
  <div
    class="fnx-picker"
  >
    <div
      class="fnx-picker__toolbar"
    />
    <div
      class="fnx-picker__columns"
      style="height: 264px;"
    >
      <div
        class="fnx-picker-column"
      >
        <ul
          class="fnx-picker-column__wrapper"
          style="transform: translate3d(0, 110px, 0);"
        >
          <li
            class="fnx-picker-column__option fnx-picker-column__option--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1990
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1991
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1992
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1993
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1994
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1995
            </div>
          </li>
        </ul>
      </div>
      <div
        class="fnx-picker__mask"
        style="background-size: 100% 110px;"
      />
      <div
        class="fnx-picker__frame fnx-hairline-unset--top-bottom"
        style="height: 44px;"
      />
    </div>
  </div>
</div>
`;

exports[`<Picker/> set picker optionHeight、visibleOptionsCount 1`] = `
<div>
  <div
    class="fnx-picker"
  >
    <div
      class="fnx-picker__toolbar"
    >
      <button
        class="fnx-picker__cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        class="fnx-picker__confirm"
        type="button"
      >
        Confirm
      </button>
    </div>
    <div
      class="fnx-picker__columns"
      style="height: 120px;"
    >
      <div
        class="fnx-picker-column"
      >
        <ul
          class="fnx-picker-column__wrapper"
          style="transform: translate3d(0, 45px, 0);"
        >
          <li
            class="fnx-picker-column__option fnx-picker-column__option--selected"
            role="button"
            style="height: 30px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1990
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 30px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1991
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 30px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1992
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 30px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1993
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 30px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1994
            </div>
          </li>
          <li
            class="fnx-picker-column__option"
            role="button"
            style="height: 30px;"
            tabindex="0"
          >
            <div
              class="fnx-ellipsis"
            >
              1995
            </div>
          </li>
        </ul>
      </div>
      <div
        class="fnx-picker__mask"
        style="background-size: 100% 45px;"
      />
      <div
        class="fnx-picker__frame fnx-hairline-unset--top-bottom"
        style="height: 30px;"
      />
    </div>
  </div>
</div>
`;
